DataFrameTransformCheckpointTests#testGetBehind fails reproducibly

Master intake: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3067/console

Reproduces locally:
```
./gradlew :x-pack:plugin:data-frame:test --tests "org.elasticsearch.xpack.dataframe.transforms.DataFrameTransformCheckpointTests.testGetBehind" \
  -Dtests.seed=828273867F1FD6F7 \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-DZ \
  -Dtests.timezone=Pacific/Niue \
  -Dcompiler.java=12 \
  -Druntime.java=11
```

```
java.lang.AssertionError: expected:<1530> but was:<1360>
        at __randomizedtesting.SeedInfo.seed([828273867F1FD6F7:48EEAF779A7DA083]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at org.elasticsearch.xpack.dataframe.transforms.DataFrameTransformCheckpointTests.testGetBehind(DataFrameTransformCheckpointTests.java:131)
```

--- a/x-pack/plugin/data-frame/src/test/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformCheckpointTests.java
+++ b/x-pack/plugin/data-frame/src/test/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformCheckpointTests.java
@@ -91,8 +91,8 @@ public class DataFrameTransformCheckpointTests extends AbstractSerializingDataFr
                 .matches(new DataFrameTransformCheckpoint(id, timestamp, checkpoint, checkpointsByIndex, (timeUpperBound / 2) + 1)));
     }
 
-    @AwaitsFix(bugUrl="https://github.com/elastic/elasticsearch/issues/41076")
     public void testGetBehind() {
+        String baseIndexName = randomAlphaOfLength(8);
         String id = randomAlphaOfLengthBetween(1, 10);
         long timestamp = randomNonNegativeLong();
 
@@ -112,7 +112,7 @@ public class DataFrameTransformCheckpointTests extends AbstractSerializingDataFr
                 checkpoints2.add(shardCheckpoint + 10);
             }
 
-            String indexName = randomAlphaOfLengthBetween(1, 10);
+            String indexName = baseIndexName + i;
 
             checkpointsByIndexOld.put(indexName, checkpoints1.stream().mapToLong(l -> l).toArray());
             checkpointsByIndexNew.put(indexName, checkpoints2.stream().mapToLong(l -> l).toArray());

